### PR TITLE
trivial: Untie efi_os_dir from the SBAT distro ID

### PR DESCRIPTION
**Pull request by hughsie**
We happily fallback with an unset -Defi_os_dir and it gets confusing explaining
why the EFI dir needs to be set on a non-supported system.

In practice they'll probably have the same eventual value on most supported
distributions.

Fixes https://github.com/fwupd/fwupd/issues/2873

Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: meson_options.txt [hunk 1]
```diff
@@ -44,6 +44,7 @@ option('efi-ldsdir', type : 'string', description : 'path to the EFI lds directo
 option('efi-includedir', type : 'string', value : '/usr/include/efi', description : 'path to the EFI header directory')
 option('efi_os_dir', type: 'string', description : 'the name of OS directory in ESP, e.g. fedora')
 option('efi_sbat_fwupd_generation', type : 'integer', value : 0, description : 'SBAT fwupd generation')
+option('efi_sbat_distro_id', type : 'string', value : '', description : 'SBAT distribution ID, e.g. fedora')
```
Comment (superm1) [inline meson_options.txt:47]: I think both `*os_dir`  values need the word `hardcoded` adjective added to them now to make it clearer why you would use them

Comment (hughsie) [reply meson_options.txt:47]: I'm not super keen changing the meson define name as that's going to break (silently?) distros using the old name to override the default -- or do you mean the description?

Comment (superm1) [reply meson_options.txt:47]: Just the description field I mean.

Comment (hughsie) [reply meson_options.txt:47]: Ohh, something like `the hardcoded name of OS directory in ESP, e.g. fedora`?

Comment (superm1) [reply meson_options.txt:47]: Exactly

Comment (hughsie) [reply meson_options.txt]: Fixed.

**Review by superm1 — APPROVED** [review]

**Pull request merged**